File: springmvc-xml-mybatis-maven/src/main/java/com/shiyan/controller/UserController.java
package com.shiyan.controller;

import com.shiyan.service.UserService;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.annotation.Resource;

/**
 * @author kevinfeng
 * @date 2019-05-25
 */
@RestController
@RequestMapping("/user")
public class UserController {
    @Resource
    private UserService userService;
}

File: springboot-mybatis-gradle-kotlin/src/main/kotlin/com/shiyan/springbootmybatisgradlekotlin/controller/UserController.kt
package com.shiyan.springbootmybatisgradlekotlin.controller

import com.easy.springboot.demo_springboot_with_mybatis.model.User
import com.easy.springboot.demo_springboot_with_mybatis.service.UserService
import com.github.pagehelper.PageInfo
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

/**
 *
 * @author kevinfeng
 * @date 2019-06-06
 */
@RestController
@RequestMapping("/user")
class UserController {
    @Autowired lateinit var userService: UserService

    @GetMapping("/findAll")
    fun findAll(): List<User> {
        return userService.findAll()
    }

    @GetMapping("/{id}")
    fun findUserById(@PathVariable(value = "id") id: Long): User {
        return userService.findUserById(id)
    }

    @GetMapping("/page")
    fun page(@RequestParam(value = "pageNo", defaultValue = "0") pageNo: Int,
             @RequestParam(value = "pageSize", defaultValue = "10") pageSize: Int): PageInfo<User> {
        return userService.page(pageNo, pageSize)
    }
}